src/Components/WebAssembly/Server/src/ComponentsWebAssemblyApplicationBuilderExtensions.cs

```diff
@@ -193,6 +215,17 @@ private static void NegotiateEncoding(HttpContext context, IWebHostEnvironment w
                 }
             }
 
+            if (StringSegment.Equals("br", selectedEncoding, StringComparison.OrdinalIgnoreCase))
+            {
+                if (ResourceExists(context, webHost, ".br"))
+                {
+                    // We only try to serve the pre-compressed file if it's actually there.
+                    context.Request.Path = context.Request.Path + ".br";
+                    context.Response.Headers[HeaderNames.ContentEncoding] = "br";
+                    context.Response.Headers.Append(HeaderNames.Vary, HeaderNames.ContentEncoding);
+                }
+            }
```

Comment:
This block pretty much exactly repeats the block above it. Could these be combined? Don't mind too much if not, it's not a big deal.